DIPs/DIP1xxy-mvf.md

```diff
@@ -0,0 +1,322 @@
+
+Walter identified the fundamental source of the problem to be the implicit conversion of integer literals to `bool`.
+
+A change to the overload resolution rules would still break code relying on the existing overload resolution behavior illustrated in [Example B](#example-b) and [Example C](#example-c), resulting in a deprecation process similar, if not the same, to that of the proposed solution, but would not close the hole in the type system.
+
+#### Disallow implicit conversion of integer literals to `bool`, but only for overloaded functions
+
+It may be possible to fix the bugs documented in [issue 9999](https://issues.dlang.org/show_bug.cgi?id=9999) and [10560](https://issues.dlang.org/show_bug.cgi?id=10560)  by preventing implicit conversion of integer literals to `bool`, but only for overloaded functions.
+
+This was rejected for similar reasons as the former; it complicates the compiler's implementation with special cases that need to be implemented, documented, explained, and understood.
+
+Also, like the former, it would still break code relying on the existing overload resolution behavior illustrated in [Example B](#example-b) and [Example C](#example-c), resulting in a deprecation process similar, if not the same, to that of the proposed solution, but would not close the hole in the type system.
+
+This solution was also not analyzed in enough detail to determine whether it would even scale.
+
+### Breaking changes / deprecation process
+
+The proposed implementation is to be rolled out through a sequence of steps to minimize any disruption.
+
+1.  If and when this DIP is approved, an announcement will be made on the D programming language forum to raise awareness of the change in the community.  Users will have a link to this DIP and the implementation so they can understand how to mitigate breakage in their code, and monitor the timing of the subsequent releases.
+
+2.  After approval of this DIP, code in DMD's test suite, Phobos, and DRuntime will be updated with the appropriate remedies documented in [Description](#description) to avoid deprecation warnings or errors in the stage 1 and stage 2 releases.  Preemptive pull requests anticipating the change proposed in this DIP have already been merged for [Phobos](https://github.com/dlang/phobos/pull/5019) and [DRuntime](https://github.com/dlang/druntime/pull/1732), but another pass will need to be made.
+
+4.  After the changes in step 2 have been pulled, stage 1 can be pulled.  From that point on the compiler will emit deprecation warnings when an implicit conversion from an integer literal to `bool` is encountered.  This will precisely identify instances in users' code that will no longer compile when stage 2 is released, but will still perform the conversion, buying users time to update their code. The deprecation will remain in place for a time period to be decided by the language authors at the time of this DIP's formal review.
```

Comment:
"been pulled, stage 1 can be pulled" => been merged, state 1 can be merged

"the language authors" => the language maintainers
That's the "official" terminology now.

"formal review" => formal assessment
Ditto.